smmregrid/util.py

```diff
@@ -38,6 +38,49 @@ def deprecated_argument(old, new, oldname='var1', newname='var2'):
             new = old
     return new
 
+
 def find_coord(ds, possible_names):
     """Find the first matching coordinate in the dataset."""
     return next((name for name in possible_names if name in ds.coords), None)
+
+
+def tolist(value):
+    """Convert value to list format. Returns [] for None, [value] for single values, unchanged for lists."""
+    if value is None:
+        return []
+    if isinstance(value, list):
+        return value
+    return [value]
+
+
+def nan_variation_check(field, time_dim, check_dims):
```

Comment:
But...if NaN vary along some vertical coordinate (I imagine some time-varying ocean grid) will they not also vary along horizontal coordinates at the same time ? Flip a NaN to not NaN somewhere and you will have changes in all three coordinates, or am I missing something of the whole idea?